python/src/iceberg/utils/schema_conversion.py

```diff
@@ -0,0 +1,454 @@
+            ...     value_is_optional=True
+            ... )
+            >>> actual == expected
+            True
+
+        Returns: A MapType
+        """
+        value_type, value_is_optional = self._resolve_union(map_type["values"])
+        return MapType(
+            key_id=map_type["key-id"],
+            # Avro only supports string keys
+            key_type=StringType(),
+            value_id=map_type["value-id"],
+            value_type=self._convert_schema(value_type),
+            value_is_optional=value_is_optional,
+        )
+
+    def _convert_logical_type(self, avro_logical_type: Dict[str, Any]) -> IcebergType:
+        """
+        Convert a schema with a logical type annotation. For the decimal and map
+        we need to fetch more keys from the dict, and for the simple ones we can just
+        look it up in the mapping.
+
+        Examples:
+            >>> from iceberg.utils.schema_conversion import AvroSchemaConversion
+            >>> avro_logical_type = {
+            ...     "type": "int",
+            ...     "logicalType": "date"
+            ... }
+            >>> actual = AvroSchemaConversion()._convert_logical_type(avro_logical_type)
+            >>> actual == DateType()
+            True
+
+        Args:
+            avro_logical_type: The logical type
+
+        Returns:
+            The converted logical type
+
+        Raises:
+            ValueError: When the logical type is unknown
+        """
+        logical_type = avro_logical_type["logicalType"]
+        physical_type = avro_logical_type["type"]
+        if logical_type == "decimal":
+            return self._convert_logical_decimal_type(avro_logical_type)
+        elif logical_type == "map":
+            return self._convert_logical_map_type(avro_logical_type)
+        elif (logical_type, physical_type) in LOGICAL_FIELD_TYPE_MAPPING:
+            return LOGICAL_FIELD_TYPE_MAPPING[(logical_type, physical_type)]
+        else:
+            raise ValueError(f"Unknown logical/physical type combination: {avro_logical_type}")
+
+    def _convert_logical_decimal_type(self, avro_type: Dict[str, Any]) -> DecimalType:
+        """
+        Args:
+            avro_type: The Avro type
+
+        Examples:
+            >>> from iceberg.utils.schema_conversion import AvroSchemaConversion
+            >>> avro_decimal_type = {
+            ...     "type": "bytes",
+            ...     "logicalType": "decimal",
+            ...     "precision": 19,
+            ...     "scale": 25
+            ... }
+            >>> actual = AvroSchemaConversion()._convert_logical_decimal_type(avro_decimal_type)
+            >>> expected = DecimalType(
+            ...     precision=19,
+            ...     scale=25
+            ... )
+            >>> actual == expected
+            True
+
+        Returns:
+            A Iceberg DecimalType
+        """
+        return DecimalType(precision=avro_type["precision"], scale=avro_type["scale"])
```

Comment:
Since you're validating physical type above, you may want to validate that the physical type here is fixed, long, or int.